docs/reference/module-api/module-overview.md

```diff
@@ -257,14 +257,231 @@ modules.export = {
+};
+```
+
+## Customization functions
+
+Customization function sections all return an object with properties that add functionality related to the module. They may add methods that can be called within the module and elsewhere, event handlers, template helpers, or other features as covered here.
+
+Each of these function sections takes the module, as `self`, as an argument. This provides access to its methods, options, and other properties, as well as those inherited from its base class.
+
+| Function name | Description |
+| ------- | ------- |
+| [`methods`](#methods-self) | Add new methods and override base class methods |
+| [`extendMethods`](#extendmethods-self) | Extend the functionality of base class methods |
+| [`components`](#components-self) | Configure asynchronous template components |
+| [`extendComponents`](#extendcomponents-self) | Extend base class template components |
+| [`helpers`](#helpers-self) | REPLACE-ME |
+| [`extendHelpers`](#extendhelpers-self) | REPLACE-ME |
+| [`apiRoutes`](#apiRoutes-self) | REPLACE-ME |
+| [`extendApiRoutes`](#extendapiroutes-self) | REPLACE-ME |
+| [`restApiRoutes`](#restapiroutes-self) | REPLACE-ME |
+| [`extendRestApiRoutes`](#extendrestapiroutes-self) | REPLACE-ME |
+| [`renderRoutes`](#renderroutes-self) | REPLACE-ME |
+| [`routes`](#routes-self) | REPLACE-ME |
+| [`handlers`](#handlers-self) | REPLACE-ME |
+| [`extendHandlers`](#extendhandlers-self) | REPLACE-ME |
+| [`queries`](#queries-self) | REPLACE-ME |
+| [`extendQueries`](#extendqueries-self) | REPLACE-ME |
+| [`middleware`](#middleware-self) | REPLACE-ME |
+| [`tasks`](#tasks-self) | REPLACE-ME |
+
+### The extension pattern
+
+Several of these sections use an extention pattern (the sections prefixed with "extend"). These sections are used to add functionality to matching features of the module's base class. Functions in `extendMethods` enhance identically named functions in the base class' `methods` section, `extendApiRoutes` enhances API routes from the base class' `apiRoutes`, and so on.
+
+Each individual function included in the extension section takes a `_super` argument in addition to the same arguments as the original function. `_super` is the original function, which should be called within the new extension.
+
+If a piece type included the `insert` in its `extendMethods` section to alter the piece titles, it might look like this:
+
+```javascript
+  extendMethods(self) {
+    return {
+      insert(_super, req, piece, options) {
+        piece.title = `🆕 ${piece.title}`;
+
+        _super(req, piece, options);
+      }
+    }
+  }
+```
+
+::: warning
+Extension functions should *always* use the `_super` argument to call the original function. If you want to *completely* overwrite the inherited function, instead add a matching function in the section without the "extend" prefix. For example, we could completely overwrite the `insert` method in our piece type by including our own `insert` function in the `methods` section.
```

Comment:
"instead add a matching function in the section without the "extend" prefix" -> "add a matching function in the section without the "extend" prefix instead".